contributor-advice.md

```diff
@@ -0,0 +1,107 @@
+# Some advice for Alpakka contributors
+
+## API design
+
+### Flows
```

Comment:
I'd move this whole `Flows` section below the `Public factory methods` where we mention the Sources, Sinks and Flows.